frontends/web/src/routes/buy/exchange.tsx

```diff
@@ -105,46 +108,17 @@ export const Exchange = ({ code, accounts }: TProps) => {
 
   }, [isPocketSupportedCoin, isMoonpaySupportedCoin, selectedRegion, regionList]);
 
-  const findBestDeal = (deals: exchangesAPI.ExchangeDeal[]): exchangesAPI.ExchangeDeal => {
-    let best = deals[0];
-    deals.forEach(deal => best = deal.fee < best.fee ? deal : best);
-    return best;
-  };
 
-  // set the bestDeal when getExchangeDeals returns.
+  // re-structure and store all exchange
+  // providers into a state to be rendered
   useEffect(() => {
     if (!exchangeDeals) {
       return;
     }
-    const pocketBest = findBestDeal(exchangeDeals.pocket);
-    const moonpayBest = findBestDeal(exchangeDeals.moonpay);
-    setBestDeal(findBestDeal([pocketBest, moonpayBest]));
-  }, [exchangeDeals]);
+    const exchanges = findBestDeal(exchangeDeals).map(ex => ({ ...ex, supported: ex.providerName === 'pocket' ? isPocketSupportedCoin : isMoonpaySupportedCoin }));
+    setAllExchangeProviders(exchanges);
+  }, [exchangeDeals, isMoonpaySupportedCoin, isPocketSupportedCoin]);
 
-  const dealsDetails = (exchange: string): string => {
-    if (!exchangeDeals) {
-      return '';
-    }
-    var details = exchange === 'pocket' ? 'Pocket' : 'Moonpay';
-    const deals = exchange === 'pocket' ? exchangeDeals.pocket : exchangeDeals.moonpay;
-    deals.forEach(deal => {
-      details += ' | ';
-      details += deal.payment === 'card' ? t('buy.exchange.creditCard') : t('buy.exchange.bankTransfer');
-      details += ' - ' + t('buy.exchange.fee') + ': ';
-      details += String(deal.fee * 100) + '%';
-      if (deal === bestDeal && showPocket && showMoonpay) {
-        details += ' - ';
-        //TODO replace with icon
-        details += 'BEST DEAL';
-      }
-      if (deal.isFast) {
-        details += ' - ';
-        //TODO replace with icon
-        details += 'FAST';
-      }
-    });
-    return details;
-  };
```

Comment:
I decided to re-structure the data using the util function I mentioned above rather than using this. There are several reasons, but it's difficult to manipulate the UI this way in react IMO. Other than that, we can't just append `details` with a JSX as it'd output a string (`[object Object]`).

Although this is good for the first version / to "prototype" the UI. 👍